torch/_inductor/codegen/cpp.py

```diff
@@ -3962,10 +3962,19 @@ def get_indexing_ranges_exprs(node):
 
                 ref_node = node2 if len(vars1) < len(vars2) else node1
 
-                extra_indexing_constraints = get_indexing_ranges_exprs(ref_node)
+                ref_indexing_constraints = get_indexing_ranges_exprs(ref_node)
 
                 node_to_recomp.recompute_size_and_body(
-                    extra_indexing_constraints=extra_indexing_constraints
+                    extra_indexing_constraints=ref_indexing_constraints
+                )
+
+                node_to_recomp_indexing_constraints = get_indexing_ranges_exprs(
+                    node_to_recomp
+                )
+
+                # fix the case where ref_node also changes its ranges
+                ref_node.recompute_size_and_body(  # type: ignore[union-attr]
```

Comment:
`recompute_size_and_body` is the method of `SchedulerNode` but not `FusedSchedulerNode`. I think we should check it in here and in the `can_fuse` checker.

Reply:
Thanks for the comment! Fixed.